sdk/core/loader/boot.cc

```diff
@@ -328,6 +338,23 @@ namespace
 		return ptr.seal(key);
 	}
 
+	template<InterruptStatus Status, typename T>
+	T *seal_return(Capability<T> ptr)
+	{
+		static_assert((Status == InterruptStatus::Enabled) ||
+		              (Status == InterruptStatus::Disabled));
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wc99-designator"
+		constexpr SealingType Sentries[] = {
+		  [int(InterruptStatus::Enabled)]  = ReturnSentryEnabling,
+		  [int(InterruptStatus::Disabled)] = ReturnSentryDisabling};
+#pragma clang diagnostic pop
```

Comment:
This makes me a bit nervous.  I'd normally do this kind of compile-time map with a template so that we get a compile error if something is invalid.

Reply:
I was just following `seal_entry` immediately above.  I'd be happy to rework both however desired, tho'.

Reply:
Yes, my code above is also bad.  I think we want something like:

```c++
template<InterruptStatus T>
constexpr SealingType SealingTypeForInterruptStatus;

template<>
constexpr SealingType SealingTypeForInterruptStatus<InterruptStatus::Enabled> = ReturnSentryEnabling;

...
```

Assuming I got the syntax right.

Reply:
Ah, no, this needs to be a dynamic thing.  Never mind.  Leave as is for now.